### ISSUE TITLE
Recursive ** expansion broken

### ISSUE THREAD

**Issue by waterhouse** [fish-shell/fish-shell#2414]
As far as I know, ** is still supposed to be recursive.  Well, it seems that it doesn't get beyond the current directory:
```
~> mkdir test
~> cd test
~/test> mkdir lol
~/test> mkdir lol/nub
~/test> touch lol/nub/y
~/test> ls -R
lol

./lol:
nub

./lol/nub:
y
~/test> echo **/y
No matches for wildcard '**/y'.
fish: echo **/y
           ^

~/test> echo $FISH_VERSION 
2.2.0-263-ga02d4dc
```
I git-bisected to find out when this happened.  There were two commits in which the build was broken, so I skipped; results are:
```
There are only 'skip'ped commits left to test.
The first bad commit could be any of:
a3f8e27bf8487cf2e74e66cabbd4b1ab6d8294a8
b55c13f275519d68ea24ecd08d4bd0f426b94722
0ac9f159d6b8235195d63f1669ca33dbc2474f47
We cannot bisect more!
```
One is named "Turn on the new wildcard expander", another is named "Eliminate recurse_to_subdirectories function", both of which seem like candidates.
I've tested this on OS X 10.10.5 and Ubuntu 14.04.

**Comment by pickfire** [comment]
I can reproduce that in Arch Linux with fish `e70ed96` as I can't figured out why I can't build fish.

I will try doing that in fish `2.2b1`.

**Comment by faho** [comment]
What works is `echo **y` - it seems the "**" _insists_ on matching the "/", i.e. it's too greedy.

**Comment by ridiculousfish** [comment]
I also noticed that e.g. `echo ~/Downloads/*/` is broken. I recently rewrote the glob expansion - looks like I messed up some cases.

**Comment by pickfire** [comment]
Yeah, `ls *` works but `ls */` doesn't works.

**Comment by ridiculousfish** [comment]
Fixed in 11376ae25bfdc90aa60511597163984e43700c22. Thanks for filing this!

**Comment by waterhouse** [comment]
Very good, confirmed working for me.
